### ISSUE TITLE
UI suggestions

### ISSUE THREAD

**Issue by spdegabrielle** [Metaxal/quickscript#61]
![image](https://user-images.githubusercontent.com/108047/120909556-00f9b880-c66e-11eb-9472-29d53ce2a496.png)

1. drop 'scripts' to make simpler menus:
manage, new, open, disable, compile and reload 

1. Unload persistent scripts -> Stop persistent scripts  ; unload sounds like they will be removed

**Comment by Metaxal** [comment]
These are interesting suggestions, thanks.
I was worried that people may confuse this with DrRacket's preferences.
(I received once a bug report for DrRacket, which is the only item that doesn't mention "scripts" :D )

But indeed there are too many "scripts".

"Stop" is more intuitive than "Unload" indeed.

**Comment by rfindler** [comment]
I think people are (rightly so!) generally confused about which repo an issue belongs in (the racket repo is particularly full of bugs that are fixed in other repos) so I think probably best to just expect to use the transfer thingy that @samth set up.

**Comment by samth** [comment]
It's not something that I set up, but a feature of GitHub for all repositories. Anyone with commit access to both can transfer issues.

**Comment by spdegabrielle** [comment]
Updated proposal: I’ve been explicit about the changes. If you accept some/all I will do a PR.

Scripts > Manage scripts => Scripts > Manage  

Scripts > Manage scripts > New script…
 => Scripts > Manage  > New

Scripts > Manage scripts > Open script…
 => Scripts > Manage  > Open

Scripts > Manage scripts > Disable scripts…
 => Scripts > Manage  > Disable

Scripts > Manage scripts > Library…
 => Scripts > Manage  > Library

Scripts > Manage scripts > Reload menu
 => Scripts > Manage  > Reload

Scripts > Manage scripts > Compile scripts and reload 
 => Scripts > Manage  > Compile and reload

Scripts > Manage scripts > Unload persistent scripts
 => Scripts > Manage  > Stop scripts

Scripts > Manage scripts > Feedback/Bug report…
 => Scripts > Manage  > Feedback/report

—-
notes: 
- I’m not not a fan of ellipsis (…) 
- The /Bug should be /bug to be consistent with the rest of the script menu - in Edge or Chrome only names are capitalised: ‘Developer tools’ , ‘Cast media to device’, ‘Maths Solver’.
- I haven’t given up on the tool palette as per VSCode, and perhaps a ribbon like ms word…

**Comment by rfindler** [comment]
The ellipsis has a rule: if the menu item opens a dialog to ask more details of what to do, there should be an ellipsis. Otherwise not.

**Comment by spdegabrielle** [comment]
Updated proposal: I’ve been explicit about the changes. If you accept some/all I will do a PR.

Scripts > Manage scripts => Scripts > Manage  

Scripts > Manage scripts > New script…
 => Scripts > Manage  > New

Scripts > Manage scripts > Open script…
 => Scripts > Manage  > Open

Scripts > Manage scripts > Disable scripts…
 => Scripts > Manage  > Disable

Scripts > Manage scripts > Library…
 => Scripts > Manage  > Library

Scripts > Manage scripts > Reload menu
 => Scripts > Manage  > Reload

Scripts > Manage scripts > Compile scripts and reload 
 => Scripts > Manage  > Compile and reload

Scripts > Manage scripts > Unload persistent scripts
 => Scripts > Manage  > Stop scripts

Scripts > Manage scripts > Feedback/Bug report…
 => Scripts > Manage  > Feedback/report

—-
notes: 
- I’m not not a fan of ellipsis (…) 
- The /Bug should be /bug to be consistent with the rest of the script menu - in Edge or Chrome only names are capitalised: ‘Developer tools’ , ‘Cast media to device’, ‘Maths Solver’.
- I haven’t given up on the tool palette as per VSCode, and perhaps a ribbon like ms word…

**Comment by Metaxal** [comment]
Counter proposal for a few ones:

Scripts > Manage scripts > Reload menu
=> Scripts > Manage > Reload menu

otherwise it may be unclear what is being reloaded. Same for:

Scripts > Manage scripts > Compile scripts and reload
=> Scripts > Manage > Compile scripts and reload

removing 'scripts' loses the plural and makes the item much more ambiguous.

Scripts > Manage scripts > Unload persistent scripts
=> Scripts > Manage > Stop persistent scripts

'persistent' is an important keyword here, as it relates to scripts that have the `#:persistent` option.

Scripts > Manage scripts > Feedback/Bug report…
=> Scripts > Manage > Feedback, bug report…

Removing 'bug' altogether reads as "If you want to report a script for being inadequate, click here".

Thoughts?

**Comment by Metaxal** [comment]
@rfindler I didn't know about the Transfer thingy, that's nice. As @samth says, it requires you to have commit access to both, and only you Robby has these rights for both Quickscript and DrRacket, so I'll probably ping you about the rare misplaced issue then :p

**Comment by rfindler** [comment]
That sounds good to me!

**Comment by spdegabrielle** [comment]
> Scripts > Manage scripts > Reload menu
> => Scripts > Manage > Reload menu

good

> otherwise it may be unclear what is being reloaded. Same for:
> 
> Scripts > Manage scripts > Compile scripts and reload
> => Scripts > Manage > Compile scripts and reload
> 
> removing 'scripts' loses the plural and makes the item much more ambiguous.
> 
> Scripts > Manage scripts > Unload persistent scripts
> => Scripts > Manage > Stop persistent scripts
> 
> 'persistent' is an important keyword here, as it relates to scripts that have the #:persistent option.

All good 😄 

> Scripts > Manage scripts > Feedback/Bug report…
> => Scripts > Manage > Feedback, bug report…
> 

> Removing 'bug' altogether reads as "If you want to report a script for being inadequate, click here".
?
Feels a bit wordy to me
How about a simple feedback/report:

Scripts > Manage scripts > Feedback/Bug report…
 => Scripts > Manage > Feedback/report…

**Comment by spdegabrielle** [comment]
Updated my response above

**Comment by Metaxal** [comment]
> > Scripts > Manage scripts > Compile scripts and reload
> > => Scripts > Manage > Compile scripts and reload
> > removing 'scripts' loses the plural and makes the item much more ambiguous.
> 
> The menu is called scripts - avoid unnecessary repetition
> => Scripts > Manage > Compile and reload

…unless it's ambiguous. Several items in the same menu are about dealing with one script only (Open, New). Some items are not even about scripts themselves (Feedback). Removing 'scripts' from 'Compile scripts and reload' makes it unclear what we are talking about: is it the current script only or all scripts?

> > Scripts > Manage scripts > Unload persistent scripts
> > => Scripts > Manage > Stop persistent scripts
> > 'persistent' is an important keyword here, as it relates to scripts that have the #:persistent option.
> 
> maybe?
> 
> => Scripts > Manage > Stop #:persistent scripts

'Stop persistent scripts' is fine.

> > Scripts > Manage scripts > Feedback/Bug report…
> > => Scripts > Manage > Feedback, bug report…
> 
> > Removing 'bug' altogether reads as "If you want to report a script for being inadequate, click here". ?
> 
> Feels a bit wordy to me
> 
> How about a simple feedback/report:
> 
> Scripts > Manage scripts > Feedback/Bug report…
> => Scripts > Manage > Feedback/report…
> 
> My ‘rule of thumb’ is that short+simple makes menu options easier to locate and parse.

"As simple as possible, but no simpler." Removing too much increases ambiguity. I prefer wordy over "What the heck is this item about?"

**Comment by rfindler** [comment]
Since these menu items will end up in the default DrRacket, may I ask that we avoid constructions of the form "x/y" and instead use actual English?

**Comment by spdegabrielle** [comment]
How about
=> Scripts > Manage > Feedback or report…

**Comment by Metaxal** [comment]
@rfindler The way I view it, you should have the last word on all Quickscript matters. So ok for plain English.

@spdegabrielle to be more in line with DrRacket, how about "Report an issue..." instead?

**Comment by Metaxal** [comment]
@rfindler @samth May I ask for your opinion on this: Do you prefer the menus to be
`Scripts | Manage Scripts | New script…`
or
`Scripts | Manage | New…`
?

I understand @spdegabrielle 's wish for less verbosity and repetition, but I'm a little worried that a number of people may just read the last item of the path and understand `New…` as in `File | New…` rather than as in `New quickscript…`

**Comment by spdegabrielle** [comment]
Lol I just remembered I promised to do a pr for the final decision

**Comment by rfindler** [comment]
I like "New Script" over "New". I think I like "Manage" over "Manage Scripts". (How's that for causing trouble? 😛 ) I have no strong opinions here.

**Comment by Metaxal** [comment]
"Manage" is right under "Scripts" so that should be fine. I also prefer New script, in particular if we go for Manage, because then the last reference of Scripts is two levels up.

How does this look?
```
Scripts 
  Manage
    New script…
    Open script…
    Disable scripts…
    ———————
    Library
    Reload menu
    Compile scripts
    Stop persistent scripts
    ———————
    Help
    Report an issue…
```

I removed "and reload" after "Compile scripts" because it's kind of expected anyway.

**Comment by spdegabrielle** [comment]
LGTM

```
Scripts > Manage scripts         => Scripts > Manage
     Compile scripts and reload  =>   Compile scripts
     Feedback/Bug report…        =>   Report an issue…
```

submitted as #62   https://github.com/Metaxal/quickscript/pull/62/files

**Comment by rfindler** [comment]
Is "Report an Issue..." going to open the github issues page? If so, then it should not have the ellipsis. Ellipses are only for when there is more information required to be able to complete the required command (not when the command's effect is to open a new window).

**Comment by spdegabrielle** [comment]
thank you @rfindler I'll fix in the PR

**Comment by Metaxal** [comment]
closed by #62
